admin/server/server.go

```diff
@@ -290,13 +309,22 @@ func checkUserAgent(ctx context.Context) (context.Context, error) {
 		return nil, status.Error(codes.PermissionDenied, fmt.Sprintf("could not parse rill-cli version: %s", err.Error()))
 	}
 
-	if v.LessThan(minCliVersion) {
-		return nil, status.Error(codes.PermissionDenied, fmt.Sprintf("Rill %s is no longer supported, please upgrade to the latest version", v))
+	if v.LessThan(minVersion) {
+		return nil, status.Error(codes.PermissionDenied, fmt.Sprintf("Rill %s is no longer supported for given operation, please upgrade to the latest version", v))
 	}
```

Comment:
I would propose extracting the method from the `ctx` here and doing the method checks here. It keeps the version checking logic contained in one func with a clear call path (the implicit way `AuthFuncOverride` gets called is pretty weird, I had to check the docs to understand why there wasn't a reference to it)